structuralcodes/materials/reinforcement/_reinforcementEC2_2004.py

```diff
@@ -71,3 +75,29 @@ def gamma_eps(self) -> float:
         # Here we should implement the interaction with the globally set
         # national annex. For now, we simply return the default value.
         return self._gamma_eps or 0.9
+
+    def __elastic__(self) -> dict:
+        """Returns kwargs for creating an elastic constitutive law."""
+        return {'E': self.Es}
+
+    def __elasticperfectlyplastic__(self) -> dict:
+        """Returns kwargs for ElasticPlastic constitutive law with no strain
+        hardening.
+        """
+        return {
+            'E': self.Es,
+            'fy': self.fyd(),
+            'eps_su': self.epsud(),
+        }
+
+    def __elasticplastic__(self) -> dict:
+        """Returns kwargs for ElasticPlastic constitutive law with strain
+        hardening.
+        """
+        Eh = (self.ftd() - self.fyd()) / (self.epsuk - self.epsyd)
```

Comment:
Should it be `self.epsud - self.epsyd` in the denominator?

Reply:
I think it is correct. Looking at the image the slope is defined when ftd is at epsuk. Then the law is interrupted at epsud. Makes sense?

![image](https://github.com/user-attachments/assets/edf70844-b444-4fce-8d6d-672a15a3e1fd)

Reply:
Makes perfect sense, thanks for double checking 😃